spec.html

```diff
@@ -737,16 +742,22 @@ <h1>DecorateConstructor ( _elements_, _decorators_ )</h1>
       <emu-alg>
         1. Let _finishers_ be a new empty List.
         1. For each _decorator_ in _decorators_, in reverse list order do
-          1. Let _privateElements_ be a List consisting of the ordered subsequence of _elements_ for each _element_ where _element_.[[Key]] is a Private Name.
-          1. Let _publicElements_ be a List consisting of the ordered subsequence of _elements_ for each _element_ where _element_.[[Key]] is not a Private Name.
-          1. Let _obj_ be FromClassDescriptor(_publicElements_).
+          1. Let _internalElements_ be an empty List.
+          1. Let _visibleElements_ be an empty List.
+          1. For each _element_ in _elements_,
+            1. If _element_.[[Kind]] is `"initializer"` or _element_.[[Key]] is a Private Name,
+              1. Add _element_ to _internalElements_.
+              1. Set _element_ to the ElementDescriptor { [[Kind]]: `"internal"`, [[Element]]: _element_ }.
+            1. Add _element_ to _visibleElements_.
+          1. Let _brand_ be a new unique Symbol value.
```

Comment:
This is the part of this PR I feel worse about: I think that the behavior is what it should be, but the implementation feels too much hacky.
Here I used a symbol (an ecmascript value), which is then stored in an internal [[Brand]] slot, to check if the element descriptors come from this decorator invocation (and thus they shoudn't be reordered).